docs/mcmc/index.rst

```diff
@@ -298,34 +318,58 @@ the results match reality::
     >>> cdf.prepare(pvals[2, :] - truth.ypos.val, r'$\Delta y$')
     >>> cdf.plot(clearwindow=False)
     >>> plt.title('')
-
-.. image:: ../_static/mcmc/mcmc_cdf_xpos.png
+    >>> plt.subplots_adjust(hspace=0.4)
 
 Scatter plot
 ------------
 
-::
+.. plot::
+    :include-source:
+    :context: close-figs
 
     >>> plt.scatter(pvals[0, :] - truth.r0.val,
     ...             pvals[4, :] - truth.alpha.val, alpha=0.3)
     >>> plt.xlabel(r'$\Delta r_0$', size=18)
-    >>> plt.ylabel(r'$\Delta \alpha$', size=18)
-
-.. image:: ../_static/mcmc/mcmc_scatter_r0_alpha.png
+    >>> _ = plt.ylabel(r'$\Delta \alpha$', size=18)
    
 This can be compared to the
-:py:class:`~sherpa.plot.RegionProjection` calculation::
+:py:class:`~sherpa.plot.RegionProjection` calculation:
 
-    >>> plt.scatter(pvals[0, :], pvals[4, :], alpha=0.3)
+.. plot::
+    :include-source:
+    :context: close-figs
 
+    >>> plt.scatter(pvals[0, :], pvals[4, :], alpha=0.3)
     >>> from sherpa.plot import RegionProjection
     >>> rproj = RegionProjection()
     >>> rproj.prepare(min=[95, 1.8], max=[150, 2.6], nloop=[21, 21])
     >>> rproj.calc(f, mdl.r0, mdl.alpha)
     >>> rproj.contour(overplot=True)
-    >>> plt.xlabel(r'$r_0$'); plt.ylabel(r'$\alpha$')
+    >>> _ = plt.xlabel(r'$r_0$'); plt.ylabel(r'$\alpha$')
```

Comment:
```suggestion
    >>> plt.xlabel(r'$r_0$')
    >>> _ = plt.ylabel(r'$\alpha$')
```

If you do need to suppress the output then I'd have thought the line should have been

```
plt.xlabel(..); _ = plt.ylabel(..)
```

(unless my understanding of how multi-line statements are handled is wrong, which is possible) but I think the need to "hide" the return value means that we should just split up these two lines.

Reply:
We should split the lines in any case, there is not reasons that needs to be on a single line.

I was hiding output in other places (and then there, but didn't even notice that I had a double statement on a single line) because printing the return values to the screen annoyed me when I played with this example. But it seems that the return values are not rendered on RTD anyway because the entire code is inside a `.. plot` directive, so it's probably more confusing for users to see `_ =` in the docs than it would be if they had some additional output when they copy the example line by line into an interpreter. 

However, our tests with doctest fail if I don't capture the return values in some way.

Reply:
Don't you just love consistent systems .....